lldb/source/Plugins/SymbolFile/DWARF/DWARFASTParserSwiftDescriptorFinder.cpp

```diff
@@ -227,11 +232,69 @@ class DWARFFieldDescriptorImpl : public swift::reflection::FieldDescriptorBase {
       bool is_indirect_case = false;
       // Unused by type info construction.
       bool is_var = false;
-      fields.emplace_back(std::make_unique<DWARFFieldRecordImpl>(
-          is_indirect_case, is_var, ConstString(member_field_name),
-          member_mangled_typename));
+     
+      // If there is a type, this case has a payload.
+      if (member_type)
+        payload_fields.emplace_back(std::make_unique<DWARFFieldRecordImpl>(
+            is_indirect_case, is_var, ConstString(member_field_name),
+            member_mangled_typename));
+      else
+        non_payload_fields.emplace_back(std::make_unique<DWARFFieldRecordImpl>(
+            is_indirect_case, is_var, ConstString(member_field_name),
+            member_mangled_typename));
     }
-    return fields;
+    // Add the non-payload cases to the end.
+    payload_fields.insert(payload_fields.end(),
+                          std::make_move_iterator(non_payload_fields.begin()),
+                          std::make_move_iterator(non_payload_fields.end()));
+    return payload_fields;
+  }
+};
+
+class DWARFMultiPayloadEnumDescriptorImpl
+    : public swift::reflection::MultiPayloadEnumDescriptorBase {
+  ConstString m_mangled_name;
+  DIERef m_die_ref;
+  std::vector<uint8_t> m_spare_bits_mask;
+  uint64_t m_byte_offset;
+
+public:
+  ~DWARFMultiPayloadEnumDescriptorImpl() override = default;
+
+  DWARFMultiPayloadEnumDescriptorImpl(ConstString mangled_name, DIERef die_ref,
+                                      std::vector<uint8_t> &&spare_bits_mask,
+                                      uint64_t byte_offset)
+      : swift::reflection::MultiPayloadEnumDescriptorBase(),
+        m_mangled_name(mangled_name), m_die_ref(die_ref),
+        m_spare_bits_mask(std::move(spare_bits_mask)),
+        m_byte_offset(byte_offset) {}
+
+  llvm::StringRef getMangledTypeName() override {
+    return m_mangled_name.GetStringRef();
+  }
+
+  uint32_t getContentsSizeInWords() const override {
+    return m_spare_bits_mask.size() / 4;
```

Comment:
This could benefit from a comment. Why 4; (is this dependent on ptrsize?) is this always divisible?

Reply:
I'm using the same terminology as `GenReflection` for the function name here. A word in this case is 32 bits long so always divisible by 4.